src/controllers/menu.controller.ts

```diff
@@ -0,0 +1,82 @@
+import * as Koa from 'koa'
+import {getRepository, Repository} from 'typeorm'
+import Dish from '../models/menu.entity'
+import * as HttpStatus from 'http-status-codes'
+import {Like} from "typeorm";
+
+export default class MenuController {
+
+    static async getMenu(ctx: Koa.Context): Promise<void> {
+        const menuRepo: Repository<Dish> = getRepository(Dish)
+        const DishName = ctx.request.query.name
+        const CategoryName = ctx.request.query.category
+        let dishes: Dish[]
+        if (ctx.request.query.name) {
+            dishes = await menuRepo.find(
+                {
+                    name: Like(`%${(DishName as string).toUpperCase()}%`)
+                }
+            )
+        } else if (ctx.request.query.category) {
+            dishes = await menuRepo.find(
```

Comment:
I would say that the better way would be constructing `where` condition depending on request params and only then call `menuRepo.find`. So that you had just one call of find method.